frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/create-obc.tsx

```diff
@@ -0,0 +1,125 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { Link } from 'react-router-dom';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { k8sCreate, K8sResourceKind, referenceFor } from '@console/internal/module/k8s';
+import { ButtonBar, history, resourceObjPath } from '@console/internal/components/utils';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import { NooBaaObjectBucketClaimModel } from '@console/noobaa-storage-plugin/src/models';
+import { match } from 'react-router';
+import { commonReducer, defaultState } from '../object-bucket-page/state';
+
+export const CreateOBCPage: React.FC<CreateOBCPageProps> = (props) => {
+  const [state, dispatch] = React.useReducer(commonReducer, defaultState);
+
+  const namespace = _.get(props, 'match.params.ns');
+
+  React.useEffect(() => {
+    const obj: K8sResourceKind = {
+      apiVersion: `${NooBaaObjectBucketClaimModel.apiGroup}/${
+        NooBaaObjectBucketClaimModel.apiVersion
+      }`,
+      kind: NooBaaObjectBucketClaimModel.kind,
+      metadata: {
+        namespace,
+      },
+      spec: {
+        ssl: false,
+      },
+    };
+    if (state.scName) {
+      obj.spec.storageClassName = state.scName;
+    }
+    if (state.name) {
+      obj.metadata.name = state.name;
+      obj.spec.generateBucketName = state.name;
+    } else {
+      obj.metadata.generateName = 'true';
+      obj.metadata.generateBucketName = true;
```

Comment:
Why `obj.metadata.generateName = 'true';` true is in quotes where as `obj.metadata.generateBucketName = true;` its a bool value.